packages/mui-material/src/Alert/Alert.d.ts

```diff
@@ -33,6 +33,22 @@ export interface AlertProps extends StandardProps<PaperProps, 'variant'> {
    * [palette customization guide](https://mui.com/material-ui/customization/palette/#adding-new-colors).
    */
   color?: OverridableStringUnion<AlertColor, AlertPropsColorOverrides>;
+  /**
+   * The components used for each slot inside the Alert.
+   * Either a string to use a HTML element or a component.
+   * @default {}
+   */
+  components?: {
+    CloseIcon?: React.ElementType;
+  };
+  /**
+   * The props used for each slot inside.
+   * @default {}
+   */
+  componentsProps?: {
+    closeIconButton?: Partial<IconButtonProps>;
```

Comment:
```suggestion
    closeButton?: Partial<IconButtonProps>;
```

Reply:
And if we truly allow other components to be used for close button, these types would need to be different.